Documentation/contributor-guide/dependency_management.md

```diff
@@ -0,0 +1,86 @@
+Dependency management
+======
+
+# Table of Contents
+- **[Main branch](#main-branch)**
+    - [Dependencies used in workflows](#dependencies-used-in-workflows)
+    - [Bumping order](#bumping-order)
+    - [Steps to bump a dependency](#steps-to-bump-a-dependency)
+    - [Indirect dependencies](#indirect-dependencies)
+    - [About gRPC](#about-grpc)
+- **[Stable branches](#stable-branches)**
+
+# Main branch
+
+The dependabot is enabled & [configured](https://github.com/etcd-io/etcd/blob/main/.github/dependabot.yml) to
+manage dependencies for etcd `main` branch. But dependabot doesn't work well for multi-module repository like `etcd`,
+see [dependabot-core/issues/6678](https://github.com/dependabot/dependabot-core/issues/6678). 
+Usually human intervention is required each time when dependabot automatically opens some PRs to bump dependencies.
+Please see guidance below.
+
+## Dependencies used in workflows
+The PRs which automatically bump dependencies (see examples below) used in workflows are fine, and can be approved & merged directly as long as all checks are successful.
+- [build(deps): bump github/codeql-action from 2.2.11 to 2.2.12](https://github.com/etcd-io/etcd/pull/15736)
+- [build(deps): bump actions/checkout from 3.5.0 to 3.5.2](https://github.com/etcd-io/etcd/pull/15735)
+- [build(deps): bump ossf/scorecard-action from 2.1.2 to 2.1.3](https://github.com/etcd-io/etcd/pull/15607)
+
+## Bumping order
+When multiple etcd modules depend on the same package, please bump the package version for all the modules in the correct order. The rule is simple:
```

Comment:
Can we please add some context on how to identify these dependencies (for example how did we hit upon : `github.com/spf13/cobra`). These typically come from code changes so will something like the following be a good prefix to identify the dependency changes

> git clone https://github.com/etcd-io/etcd.git 
> go mod tidy  //check output
>

Reply:
Please read the paragraph right below` # Main branch`

Reply:
```suggestion
When multiple etcd modules depend on the same package, please close the dependabot PRs and bump the package version for all the modules in the correct order. The rule is simple:
```
I was looking at the earlier fix (https://github.com/etcd-io/etcd/pull/15313) where dependabot changes were being restricted to root and thought we won't be getting these PRs anymore. Is it incumbent on the person raising the request to close the dependabot PRs ?

Reply:
yes, the related PRs opened by dependabot need to be closed. Updated the section "## Steps to bump a dependency". thx